op-service/sources/l2_client.go

```diff
@@ -197,11 +201,14 @@ func (s *L2Client) outputV0(ctx context.Context, block eth.BlockInfo) (*eth.Outp
 
 	blockHash := block.Hash()
 	var messagePasserStorageRoot eth.Bytes32
-	if s.rollupCfg.IsIsthmus(block.Time()) {
+	if s.rollupCfg.IsIsthmus(block.Time()) && !s.fetchWithdrawalRootFromState {
+		s.log.Debug("Retrieving messagePasserStorageRoot from block header")
 		// If Isthmus hard fork has activated, we can get the messagePasserStorageRoot directly from the header
 		// instead of having to compute it from the contract storage trie.
 		messagePasserStorageRoot = eth.Bytes32(*block.WithdrawalsRoot())
+
 	} else {
+		s.log.Debug("Bypassing block header, retrieving messagePasserStorageRoot from state")
```

Comment:
wondering if we also want to call it withdrawals root in the logs, even though the variable has the less common name?

Reply:
[8049f7f](https://github.com/ethereum-optimism/optimism/pull/15150/commits/8049f7f156f2c90c9c697d21e7dd7fc3ec2d02c1)